modules/http/src/main/scala/higherkindness/mu/http/Utils.scala

```diff
@@ -0,0 +1,124 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package higherkindness.mu.http
+
+import cats.ApplicativeError
+import cats.effect._
+import cats.implicits._
+import fs2.interop.reactivestreams._
+import fs2.{RaiseThrowable, Stream}
+import io.grpc.Status.Code._
+import org.typelevel.jawn.ParseException
+import io.circe._
+import io.circe.generic.auto._
+import io.circe.jawn.CirceSupportParser.facade
+import io.circe.syntax._
+import io.grpc.{Status => _, _}
+import jawnfs2._
+import monix.execution._
+import monix.reactive.Observable
+import org.http4s._
+import org.http4s.dsl.Http4sDsl
+import scala.concurrent.ExecutionContext
+import scala.util.control.NoStackTrace
+
+object Utils {
+
+  implicit class MessageOps[F[_]](val message: Message[F]) extends AnyVal {
+
+    def jsonBodyAsStream[A](
+        implicit decoder: Decoder[A],
+        F: ApplicativeError[F, Throwable]): Stream[F, A] =
+      message.body.chunks.parseJsonStream.map(_.as[A]).rethrow
+  }
+
+  implicit class RequestOps[F[_]](val request: Request[F]) {
+
+    def asStream[A](implicit decoder: Decoder[A], F: ApplicativeError[F, Throwable]): Stream[F, A] =
+      request
+        .jsonBodyAsStream[A]
+        .adaptError { // mimic behavior of MessageOps.as[T] in handling of parsing errors
+          case ex: ParseException =>
+            MalformedMessageBodyFailure(ex.getMessage, Some(ex)) // will return 400 instead of 500
+        }
+  }
+
+  implicit class ResponseOps[F[_]](val response: Response[F]) {
+
+    implicit private val throwableDecoder: Decoder[Throwable] =
+      Decoder.decodeTuple2[String, String].map {
+        case (cls, msg) =>
+          Class
+            .forName(cls)
+            .getConstructor(classOf[String])
+            .newInstance(msg)
+            .asInstanceOf[Throwable]
+      }
+
+    def asStream[A](
+        implicit decoder: Decoder[A],
+        F: ApplicativeError[F, Throwable],
+        R: RaiseThrowable[F]): Stream[F, A] =
+      if (response.status.code != 200) Stream.raiseError(ResponseError(response.status))
+      else response.jsonBodyAsStream[Either[Throwable, A]].rethrow
+  }
+
+  implicit class Fs2StreamOps[F[_], A](stream: Stream[F, A]) {
+
+    implicit private val throwableEncoder: Encoder[Throwable] = new Encoder[Throwable] {
+      def apply(ex: Throwable): Json = (ex.getClass.getName, ex.getMessage).asJson
+    }
+
+    def asJsonEither(implicit encoder: Encoder[A]): Stream[F, Json] = stream.attempt.map(_.asJson)
+
+    def toObservable(implicit F: ConcurrentEffect[F], ec: ExecutionContext): Observable[A] =
```

Comment:
I wouldn't offer this combinator, in the past we used it and they turned out unsafe.

Reply:
Removed `toObservable` at https://github.com/higherkindness/mu/pull/552/commits/c922e0f1da911b7a61752907dc882e8522eae495